### PR TITLE
[Hub Generated] Publish private branch 'user/jimmyca/fixAppConfigSpec'

### PR DESCRIPTION
**Pull request by jimmyca15**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/published%2Fjimmyca15%2Fazure-rest-api-specs-pr%2Fuser%2Fjimmyca%2FfixAppConfigSpec...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by jimmyca15** [comment]
The PR is created based on the updates in the private branch. The updates in the PR has already been reviewed and approved with this PR [Azure/azure-rest-api-specs-pr/7780](https://github.com/Azure/azure-rest-api-specs-pr/pull/7780)

**Comment by jimmyca15** [comment]
All breaking changes have been approved in the private repo PR.

**Comment by jimmyca15** [comment]
Hello @qiaozha please let me know if there is anything I need to do to complete this PR. All approvals, including breaking change approval can be found in the private branch PR that this was published from.

**Comment by jimmyca15** [comment]
Hello @qiaozha 

The breaking changes for this were approved in the private repo. can we merge this?

**Comment by qiaozha** [comment]
@jimmyca15 Can you fix the avocado failure and prettier check failure ?

**Comment by qiaozha** [comment]
The api breaking change is approved in previous private repo PR.

File: specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/appconfiguration.json [hunk 1]
```diff
@@ -812,58 +816,6 @@
         }
       }
     },
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppConfiguration/configurationStores/{configStoreName}/keyValues": {
```
Comment (ArcturusZhang) [inline specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2022-05-01/appconfiguration.json:815]: Why we remove this operation?
This is not only a breaking change for the swagger and all SDKs, but also breaks the ARM linter rule that a resource must have a list operation on its parent

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

**Review by ArcturusZhang — APPROVED** [review]
I saw there are the reasons in the corresponding private PR. Approved

**Comment by jimmyca15** [comment]
@qiaozha

I have fixed Avocado and Prettier failures.

**Comment by jimmyca15** [comment]
@ArcturusZhang @qiaozha

This was approved but never merged. Can you merge it?

**Comment by jimmyca15** [comment]
@ArcturusZhang @qiaozha

This was approved but never merged. Can you merge it?

**Comment by jimmyca15** [comment]
@ArcturusZhang @qiaozha

This was approved but never merged. Can you merge it?

**Comment by qiaozha** [comment]
@jimmyca15 can you fix the swagger model validation errors ?

**Comment by qiaozha** [comment]
Also, there's staging lint diff error

**Comment by jimmyca15** [comment]
@qiaozha Wouldn't fixing those model validation errors be an unapproved breaking change?

Also, the lint diff error is the approved breaking change this PR is making. Is there a way for me to silence that lint diff error?

File: specification/appconfiguration/resource-manager/readme.md [hunk 2]
```diff
@@ -176,6 +176,12 @@ directive:
     from: appconfiguration.json
     where: $.definitions.OperationDefinition.properties.isDataAction
     reason: This is a standardized ARM API.
+  - suppress: R4015
+    from: appconfiguration.json
+    where: $.definitions.KeyValue
+  - suppress: R3010
+    from: appconfiguration.json
+    where: $.definitions.KeyValue
```
Comment (qiaozha) [inline specification/appconfiguration/resource-manager/readme.md:184]: @jianyexi Can you help review the suppression directive here ? Thanks

**Comment by qiaozha** [comment]
@ArcturusZhang Can you help take a look at the dotnet sdk track2 generation error ? Thanks

**Comment by jimmyca15** [comment]
@qiaozha I see you mentioned some people. Should I be reaching out to these individuals for the concerns you mentioned? We really need this checked in.

Please note that all these breaking changes were requested and approved in the private repo PR this was generated from.

https://github.com/Azure/azure-rest-api-specs-pr/pull/7780

File: specification/appconfiguration/resource-manager/readme.md [hunk 3]
```diff
@@ -176,6 +176,12 @@ directive:
     from: appconfiguration.json
     where: $.definitions.OperationDefinition.properties.isDataAction
     reason: This is a standardized ARM API.
+  - suppress: R4015
```
Comment (jianyexi) [inline specification/appconfiguration/resource-manager/readme.md:179]:
```suggestion
  - suppress: NestedResourcesMustHaveListOperation

File: specification/appconfiguration/resource-manager/readme.md [hunk 4]
```diff
@@ -176,6 +176,12 @@ directive:
...
+    from: appconfiguration.json
+    where: $.definitions.KeyValue
+  - suppress: R3010
```
Comment (jianyexi) [inline specification/appconfiguration/resource-manager/readme.md:182]:
```suggestion
  - suppress: TrackedResourceListByImmediateParent
```

**Comment by raych1** [comment]
@jimmyca15 , please refer to this [example](https://github.com/Azure/azure-rest-api-specs/blob/5e638554b6/specification/appplatform/resource-manager/Microsoft.AppPlatform/stable/2020-07-01/examples/Apps_Update.json#L64) to add LRO headers to swagger examples to fix `model validation` errors.

**Comment by jimmyca15** [comment]
@raych1 I have added it and resolved the model validation errors.

File: specification/appconfiguration/resource-manager/readme.md (hunk 4 above)
Comment (jimmyca15) [reply specification/appconfiguration/resource-manager/readme.md]: Updated.

File: specification/appconfiguration/resource-manager/readme.md (hunk 3 above)
Comment (jimmyca15) [reply specification/appconfiguration/resource-manager/readme.md]: Updated.

**Review by qiaozha — APPROVED** [review]

**Pull request merged**